src/map/homunculus.cpp

```diff
@@ -280,29 +276,30 @@ int hom_delete(struct homun_data *hd, int emote)
 * @param flag_envolve
 */
 void hom_calc_skilltree(struct homun_data *hd, bool flag_evolve) {
-	uint8 i;
-	short c = 0;
-
 	nullpo_retv(hd);
 
+	int32 c = hom_class2index(hd->homunculus.prev_class);
```

Comment:
`homunculus_db` should use the class id as key (not the index from hom_class2index)
```suggestion
	auto homun = homunculus_db.find(hd->homunculus.prev_class);
```